### PR TITLE
Developer uses extension - feature

### PR DESCRIPTION
**Pull request by gajdaw**
As of now, I belive this is not covered by tests.

### CODE REVIEW

**Comment by stof** [comment]
AFAICT, these scenarios are not testing anything. If we disable the loading of extensions in PhpSpec, your scenarios will still pass as there is nothing relying on the extension in the suite you are running

File: features/extensions/developer_uses_extension.feature [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+    use PhpSpec\Matcher\IdentityMatcher;
+
+    class Extension implements ExtensionInterface
+    {
+        /**
+         * @param ServiceContainer $container
+         */
+        public function load(ServiceContainer $container)
+        {
+          $container->set('matchers.anotherIdentityMatcher', function (ServiceContainer $c) {
```
Comment (stof) [inline features/extensions/developer_uses_extension.feature]: wrong indentation

**Comment by ciaranmcnulty** [comment]
I agree with @stof that this test doesn't really show that anything works.

The real test is a real extension passing or failing. Perhaps we should be doing some sort of large build against known/supported extensions before a major release?

**Comment by gajdaw** [comment]
Thanks for you comments. I haven't thought about using the extension. But, of course, you are right. Including an extension in config doesn't call any methods defined in the extension.

I included a new test case `it_should_succeed_in_using_new_matcher()` that uses the new matcher defined in the extension. This required the complete implementation of a new matcher.

**Comment by stof** [comment]
this scenario is covering that extensions can provide matchers. This is what its description should be

**Comment by gajdaw** [comment]
@stof Done.

**Comment by everzet** [comment]
:+1: good job

**Pull request merged**